loader/directory_refresher.go

```diff
@@ -3,17 +3,40 @@ package loader
 import "path/filepath"
 
 type DirectoryRefresher struct {
-	currDir string
+	currDir  string
+	watchOps map[FileSystemOp]struct{}
+}
+
+var defaultFileSystemOps = map[FileSystemOp]struct{}{
+	Write:  {},
+	Create: {},
+	Chmod:  {},
 }
 
 func (d *DirectoryRefresher) WatchDirectory(runtimePath string, appDirPath string) string {
 	d.currDir = filepath.Join(runtimePath, appDirPath)
 	return d.currDir
 }
 
+func (d *DirectoryRefresher) WatchFileSystemOps(fsops ...FileSystemOp) map[FileSystemOp]struct{} {
```

Comment:
What's the expectation with what a caller may want to do with the return value? 

Nit: my suggestion would be to drop it. I see this API as the way we're asking users to specify their desired ops. But if we return the map to the caller, they can continue to modify the map outside of these calls.